src/status_im2/contexts/quo_preview/share/qr_code.cljs

```diff
@@ -1,55 +1,105 @@
+                            :size                250
+                            :avatar              :none
+                            :profile-picture     (resources/get-mock-image :user-picture-male5)
+                            :customization-color :army
+                            :full-name           "Full Name"
+                            :emoji               "🍒"
+                            :picture             (resources/get-mock-image :community-logo)
+                            :f-name              "First Name"
+                            :l-name              "Last Name"})]
     (fn []
-      (reset! media-server-uri (image-server/get-qr-image-uri-for-any-url
-                                {:url         @text
-                                 :port        (rf/sub [:mediaserver/port])
-                                 :error-level @error-correction-level
-                                 :qr-size     250}))
-      [preview/preview-container
-       {:state      state
-        :descriptor descriptor}
-       [rn/view {:style {:padding-bottom 150}}
-        [rn/view {:style {:flex 1}}]
-        [rn/view
-         {:style {:padding-vertical 60
-                  :flex-direction   :row
-                  :justify-content  :center}}
-         [rn/view
-          [quo/qr-code
-           {:source {:uri @media-server-uri}
-            :height 250
-            :width  250}]
-
-          [rn/view
-           [rn/text {:style {:padding 20 :flex-shrink 1}} "Media server url -> "
-            @media-server-uri]]]]]])))
+      (let [qr-media-server-uri (image-server/get-qr-image-uri-for-any-url
+                                 {:url         (:url @state)
+                                  :qr-size     (:size @state)
+                                  :port        media-server-port
+                                  :error-level :highest})]
+        [preview/preview-container
+         {:component-container-style {:flex            1
+                                      :justify-content :center
+                                      :align-items     :center
+                                      :margin-vertical 12}
+          :state                     state
+          :descriptor                (concat descriptor
+                                             (case (:avatar @state)
+                                               :profile        profile-descriptor
+                                               :wallet-account wallet-account-descriptor
+                                               :channel        channel-descriptor
+                                               :saved-address  saved-address-descriptor
+                                               nil))}
+         [quo/qr-code
+          (cond-> @state
+            :always
+            (assoc :qr-image-uri qr-media-server-uri)
+
+            ;; `:channel` variant receives colors as hex strings instead of keywords
```

Comment:
this comment is a bit unnecessary imo and suggests something may be wrong here 🤔 
The custom-color function should take care of this so the component in question takes in either a keyword or a hex string and that custom-colors method should handle the resolving 👍

Reply:
same for all of the components using custom colors really, it should be abstracted away by the colors mechanism

Reply:
oh yes, I totally agree, this is due to the channel-avatar implementation:
(ns: `quo2.components.avatars.channel-avatar.style`)
![image](https://github.com/status-im/status-mobile/assets/90291778/434ba817-5bef-4d1e-ae9d-b76b54c9eb26)

I think we should standarize the color mechanism according to your idea :+1: 

About the comment, I think the code is a bit weird because of this special case, so I decided to explan a little more about it.

Reply:
Btw the custom color function will currently take a hex and use that 👍